src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp

```diff
@@ -0,0 +1,654 @@
+
+graph_api::graph_api(const graph_config* pConfig)
+    : pGraphConfig(pConfig), isConfigured(false), isRunning(false), startTime(0)
+{}
+
+err_code graph_api::configure()
+{
+    if (!pGraphConfig)
+        return errorMsg(err_code::internal_error, "ERROR: adf::graph_api::configure: Invalid graph configuration.");
+
+    int numCores = pGraphConfig->coreColumns.size();
+    if (pGraphConfig->coreRows.size() != numCores || pGraphConfig->iterMemAddrs.size() != numCores
+        || pGraphConfig->triggered.size() != numCores || pGraphConfig->iterMemColumns.size() != numCores
+        || pGraphConfig->iterMemRows.size() != numCores)
+        return errorMsg(err_code::internal_error, "ERROR: adf::graph_api::configure: inconsistent number of cores.");
+
+    coreTiles.resize(numCores);
+    iterMemTiles.resize(numCores);
+    for (int i = 0; i < numCores; i++)
+    {
+        size_t numReservedRows = config_manager::s_num_reserved_rows;
+        coreTiles[i] = XAie_TileLoc(pGraphConfig->coreColumns[i], pGraphConfig->coreRows[i] + numReservedRows + 1);
+        iterMemTiles[i] = XAie_TileLoc(pGraphConfig->iterMemColumns[i], pGraphConfig->iterMemRows[i] + numReservedRows + 1);
+    }
+
+    isConfigured = true;
+    return err_code::ok;
+}
+
+err_code graph_api::run()
+{
+    if (!isConfigured)
+        return errorMsg(err_code::aie_driver_error, "ERROR: adf::graph::run: graph is not configured.");
+
+    int driverStatus = AieRC::XAIE_OK; //0
+    int numCores = coreTiles.size();
+
+    // Record a snapshot of the graph cores startup/enable time
+    if (numCores)
+        driverStatus |= XAie_ReadTimer(config_manager::s_pDevInst, coreTiles[0], XAIE_CORE_MOD, (u64*)(&startTime));
+
+    infoMsg("Enabling core(s) of graph " + pGraphConfig->name);
+
+    if (config_manager::s_broadcast_enable_core)
+    {
+        for (int i = 0; i < numCores; i++)
+        {
+            //Set Enable_Event bits to 113
+            //XAie_CoreConfigureEnableEvent(config_managers_pDevInst, coreTiles[i], XAIE_EVENT_BROADCAST_7_CORE);
+            XAie_Write32(config_manager::s_pDevInst, (_XAie_GetTileAddr(config_manager::s_pDevInst, coreTiles[i].Row, coreTiles[i].Col) + 0x00032008), 0x4472);
+        }
+
+        //Trigger event 113 in shim_tile at column 0 by writing to Event_Generate
+        XAie_EventGenerate(config_manager::s_pDevInst, XAie_TileLoc(0, 0), XAIE_PL_MOD, XAIE_EVENT_BROADCAST_A_6_PL);
+
+        for (int i = 0; i < numCores; i++)
+        {
+            //Set Enable_Event bits to 0
+            //if (XAie_Read32(config_manager::s_pDevInst, (_XAie_GetTileAddr(config_manager::s_pDevInst, coreTiles[i].Row, coreTiles[i].Col) +  0x00032004) & 0x0001) == 1)
+            //XAie_CoreConfigureEnableEvent(config_manager::s_pDevInst, coreTiles[i], XAIE_EVENT_NONE_CORE);
+            XAie_Write32(config_manager::s_pDevInst, (_XAie_GetTileAddr(config_manager::s_pDevInst, coreTiles[i].Row, coreTiles[i].Col) + 0x00032008), 0x4400);
```

Comment:
suggestion: it would be better that instant number, address can be defined with a macro something like
#define AIE_CORE_BASE_ADDR 0x00032008
#define AIE_CORE_STATUS_RESET 0x00000001
#define AIE_CORE_STATUS_ENABLE 0x00000002
#define AIE_CORE_STATUS_HALT  0x00000004

Reply:
We will address this in the following pull request